envi/memory.py

```diff
@@ -580,6 +604,58 @@ def readMemory(self, va, size, _origva=None):
 
         raise envi.SegmentationViolation(va, msg)
 
+    def _reqProbeMem(self, va, size, perm):
+        '''
+        Calls probeMemory() and either returns or raises the appropriate exception
+        '''
+        msg = None
+
+        # check starting address map
+        startmap = self.getMemoryMap(va)
+        if startmap is None:
+            msg = "Bad Memory Access (invalid memory range): 0x%x: 0x%x (%s)" % (
+                    va, size, getPermName(perm))
+            raise envi.SegmentationViolation(va, msg)
+
+        # check ending address map
+        endmap = self.getMemoryMap(va+size-1)
+        if endmap is None:
+            msg = "Bad Memory Access (invalid memory range): 0x%x: 0x%x (%s)" % (
+                    va, size, getPermName(perm))
+            raise envi.SegmentationViolation(va, msg)
+
+        # check that all memory is valid and correct permissions
+        ptr = va
+        curmap = startmap
+        endmapva, endmapsize, endmapperm, endmapfile = endmap
+
+        while True:
+            mapva, mapsize, mapperm, mapfile = curmap
+
+            # check permissions
+            if mapperm & perm != perm:
```

Comment:
This probably needs to honor self._supervisor

Reply:
done